src/osp/Active/SysVehicle.cpp

```diff
@@ -500,65 +499,21 @@ void SysVehicle::update_vehicle_modification(ActiveScene& rScene)
 
             };
 
-            std::vector<ActiveEnt> &parts
+            std::vector<ActiveEnt> &rParts
                     = view.get<ACompVehicle>(vehicleEnt).m_parts;
 
-            parts.erase(std::remove_if(parts.begin(), parts.end(),
-                                       removeDestroyed), parts.end());
+            rParts.erase(std::remove_if(rParts.begin(), rParts.end(),
+                                        removeDestroyed), rParts.end());
 
-            // update or create rigid bodies. also set center of masses
+            // Calculate transforms and center of masses of new islands
 
             for (ActiveEnt islandEnt : islands)
             {
-                auto &islandVehicle = view.get<ACompVehicle>(islandEnt);
-                auto &islandTransform
-                        = rScene.reg_get<ACompTransform>(islandEnt);
-
-                Vector3 comOffset;
-                //float totalMass;
+                auto &rIslandVehicle = view.get<ACompVehicle>(islandEnt);
+                auto &rIslandTf = rScene.reg_get<ACompTransform>(islandEnt);
```

Comment:
This whole forloop is just two getters now?

that doesn't seem like what you intended?

Reply:
Yep, this is excess code from removed center of mass calculations.